lib/heartcheck.rb

```diff
@@ -16,6 +16,9 @@ class << self
     # @attr [Heartcheck::Executors::Base] the checks executor backend
     attr_accessor :executor
 
+    # @attr [Boolean] the option for using a hash response or an array response
+    attr_reader :response_formatter
```

Comment:
I think it is not so clear if `true` would use a hash or an array response.

What about renaming it to `format_response_as_hash` or `hash_response`, then it would be used like this:

```ruby
h.hash_response = true
```

Reply:
Looking at the code below, another suggestion to the name would be `hash_formatter`

Reply:
Done!